enterprise/backend/src/metabase_enterprise/sandbox/query_processor/middleware/row_level_restrictions.clj

```diff
@@ -257,11 +257,38 @@
     (qp.store/cached card-id
                      (query-perms/required-perms (:dataset-query (lib.metadata.protocols/card (qp.store/metadata-provider) card-id))
                                                  :throw-exceptions? true))
-    {:perms/data-access (zipmap (sandbox->table-ids sandbox) (repeat :unrestricted))}))
+    {:perms/view-data :unrestricted
+     :perms/create-queries (zipmap (sandbox->table-ids sandbox) (repeat :query-builder))}))
+
+(defn- merge-perms
```

Comment:
Nice doc string, this makes sense to me. I guess previously the merge behavior didn't matter since `:data-access` would only be at the table level and `:native-query-editing` only at the DB-level for sandboxes. But now `:create-queries` could be either, depending on whether the sandbox involves a card?